File: Code coach/Medium/Military time/military_time.py
import re

time = input()
dayPeriod = ""

# position 0 because we know that it will be the first occurrence
matchedDayPeriod = re.findall(r"\s[a-zA-Z]{2}", time)[0]
if matchedDayPeriod:
	time = re.sub(matchedDayPeriod, '', time) 
	dayPeriod = matchedDayPeriod.strip().upper()  # filtered

timeSplitted = time.split(":")
hours = timeSplitted[0]
minutes = timeSplitted[1]

if len(hours) == 1:
	hours = "0{0}".format(hours)

if dayPeriod == "PM":
	hours = int(hours) + 12
elif dayPeriod != "AM":
	if int(hours) > 12:
		hours = int(hours) - 12
		minutes += " PM"
	else:
		minutes += " AM"

print(f"{hours}:{minutes}")

File: Code coach/Medium/Military time/military_time.cpp
#include <iostream>
#include <regex>
using namespace std;

int main()
{
	string time;
	getline(cin, time);

	string dayPeriod = "";

	regex exp("\\s[a-zA-Z]{2}");
	smatch res;
	string matchedDayPeriod = "";

// esto no vca
	string::const_iterator searchStart(time.cbegin());
	while (regex_search(searchStart, time.cend(), res, exp))
	{
		matchedDayPeriod += (searchStart == time.cbegin() ? "" : " ");
		matchedDayPeriod += res[0];
		// searchStart = res.suffix().first;

		break;
	}

	if (!matchedDayPeriod.empty())
	{
		size_t startPositionToErase = time.find(matchedDayPeriod);
		time.erase(startPositionToErase, matchedDayPeriod.length());
		for (auto &c : matchedDayPeriod) 
		{
			if (isalpha(c)) // avoid whitespaces
			{
				dayPeriod += toupper(c);
			}
		}
	}

	regex reg(":");
	sregex_token_iterator iter(time.begin(), time.end(), reg, -1);
	sregex_token_iterator end;
	vector<string> timeSplitted(iter, end);

	string hours = timeSplitted[0];
	string minutes = timeSplitted[1];

	if (hours.length() == 1)
	{
		hours = "0" + hours;
	}
	int hoursInt = stoi(hours);

	if (dayPeriod == "PM")
	{
		hours = to_string(hoursInt + 12);
	}
	else if (dayPeriod != "AM")
	{
		if (hoursInt > 12)
		{
			hours = to_string(hoursInt - 12);
			minutes += " PM";
		}
		else 
		{
			minutes += " AM";
		}
	}
	
	cout << hours << ":" << minutes;

	return 0;
}
